PubSub.Core/Consumers/RabbitMQConsumer.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Text;
+using PubSub.Core.Managers;
+using PubSub.Core.Services.Notifier;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+
+namespace PubSub.Core.Consumers
+{
+    public class RabbitMQConsumer : IConsumer
+    {
+        private readonly IConnectionManager _connectionManager;
+        private readonly ISubscriberNotifierService _subscriberNotifierService;
+        private readonly string _queueName;
+        private readonly HashSet<string> _callbackUrls = new();
+        private IConnection _connection;
+        private IChannel _channel;
+        private AsyncEventingBasicConsumer _consumer;
+
+
+        public RabbitMQConsumer(IConnectionManager connectionManager, ISubscriberNotifierService subscriberNotifierService, string queueName)
+        {
+            _connectionManager = connectionManager;
+            _subscriberNotifierService = subscriberNotifierService;
+            _queueName = queueName;
+        }
+
+        public virtual async Task StartListeningAsync()
```

Comment:
I don't think this method needs the virtual keyword since it's not being extended in any derived class